utils.test.js

```diff
@@ -53,4 +94,250 @@ describe(_startCase(filename), function () {
+        port: '80'
+      };
+
+      expect(fn(site)).toEqual('http://foo.com/_auth');
+    });
+  });
+
+  describe('removePrefix', function () {
+    const fn = lib[this.description];
+
+    it('removes the prefix token and anything before it', function () {
+      expect(fn('foo/bar', '/')).toEqual('bar');
+    });
+
+    it('returns the same string if the prefix token is not found', function () {
+      expect(fn('foo-bar', '/')).toEqual('foo-bar');
+    });
+  });
+
+  describe('generateStrategyName', function () {
+    const fn = lib[this.description];
+
+    it('generates a string with provider and slug', function () {
+      const site = { slug: 'nymag' };
+
+      expect(fn('twitter', site)).toEqual('twitter-nymag');
+    });
+  });
+
+  describe('removeQueryString', function () {
+    const fn = lib[this.description];
+
+    it('removes the queryString from path url', function () {
+      expect(fn('http://nymag.com?edit=true')).toEqual('http://nymag.com');
+    });
+
+    it('returns the same path url if there are no queryStrings', function () {
+      expect(fn('http://nymag.com')).toEqual('http://nymag.com');
+    });
+  });
+
+  describe('removeExtension', function () {
+    const fn = lib[this.description];
+
+    it('removes the extension from the url path', function () {
+      expect(fn('http://nymag.com/_pages/homepage.html')).toEqual('http://nymag.com/_pages/homepage');
+      expect(fn('http://nymag.com/_pages/homepage.html?edit=true')).toEqual('http://nymag.com/_pages/homepage');
+    });
+
+    it('returns the same path url if there are no extensions', function () {
+      expect(fn('http://nymag.com/_pages/homepage')).toEqual('http://nymag.com/_pages/homepage');
+    });
+
+    it('removes the extension if it does not have a leading slash', function () {
+      expect(fn('homepage.html')).toEqual('homepage');
+    });
+  });
+
+  describe('getUri', function () {
+    const fn = lib[this.description];
+
+    it('returns a formatted uri', function () {
+      const req = {
+        hostname: 'foo.com',
+        baseUrl: '/bar',
+        path: '/pizza'
+      };
+
+      expect(fn(req)).toEqual('foo.com/bar/pizza');
+    });
+  });
+
+  describe('getProviders', function () {
+    const fn = lib[this.description];
+
+    it('removes apikey and local providers', function () {
+      const providers = ['apikey', 'local', 'google'],
+        site = {
+          prefix: 'foo.com',
+          protocol: 'http',
+          port: 80
+        },
+        expected = [{
+          name: 'google',
+          url: 'http://foo.com/_auth/google',
+          title: 'Log in with Google',
+          icon: () => {}
+        }],
+        results = fn(providers, site);
+
+      expect(results.length).toEqual(1);
+      expect(results[0].url).toEqual(expected[0].url);
+    });
+  });
+
+  describe('verify', function () {
+    const fn = lib[this.description];
+
+    it('throws if there is no username field', function () {
```

Comment:
What do you think about using a beforeEach function to group?

```js
properties = { username: 'username' };
profile = { username: '' }, // maybe
req = {},
next = jest.fn(),
```

And just do modifications in the tests you want to have different situations